secure/storage/src/tests/vault.rs

```diff
@@ -6,42 +6,174 @@ use crate::{
-/// `docker/vault/run.sh`
-// There can only be one vault test as reset called on one instance can break another test
+/// Storage data constants for testing purposes.
+const U64_KEY_1: &str = "U64 Key 1";
+const U64_KEY_2: &str = "U64 Key 2";
+const U64_KEY_3: &str = "U64 Key 3";
+const U64_VALUE_1: u64 = 10;
+const U64_VALUE_2: u64 = 304;
+const U64_VALUE_3: u64 = 767;
+
+/// This holds the canonical list of vault storage tests. This is required because vault tests
+/// cannot currently be run in parallel, as each test uses the same vault instance and
+/// storage resets can interfere between tests. To avoid this, we run each test sequentially, and
+/// reset the storage engine only after each test.
+const VAULT_TESTS: &[fn()] = &[
+    test_suite_multiple_namespaces,
+    test_suite_no_namespaces,
+    test_vault_distinct_namespaces,
+    test_vault_namespace_reset,
+];
+
+/// A test for verifying VaultStorage properly implements the LibraSecureStorage API and enforces
+/// strict separation between unique namespaces. This test depends on running Vault, which can be
+/// done by using the provided docker run script in `docker/vault/run.sh`
 #[test]
 #[ignore]
 fn execute_storage_tests_vault() {
-    let mut storage = VaultStorage::new(VAULT_HOST.into(), VAULT_ROOT_TOKEN.into(), None);
-    storage.reset_and_clear().unwrap();
+    let mut storage = create_initialized_vault_storage(None);
+    storage
+        .reset_and_clear()
+        .expect("Failed to reset VaultStorage after creation!");
 
-    test_vault(&mut storage);
-    suite::execute_all_storage_tests(&mut storage);
+    for test in VAULT_TESTS.iter() {
+        test();
+        storage
+            .reset_and_clear()
+            .expect("Failed to reset storage engine between tests!");
+    }
+}
+
+/// Verifies that when calling reset on a VaultStorage instance, if the instance was created with
+/// a namespace, only the secrets within the namespace are removed. Otherwise, the entire storage
+/// instance is reset. This helps to ensure operations across namespaces do not interfere.
+fn test_vault_namespace_reset() {
+    // Create two distinct VaultStorage namespaces with difference pieces of data
+    let mut storage_1 = create_initialized_vault_storage(Some(VAULT_NAMESPACE_1.into()));
+    let mut storage_2 = create_initialized_vault_storage(Some(VAULT_NAMESPACE_2.into()));
+    storage_1
+        .create(U64_KEY_1, Value::U64(U64_VALUE_1), &Policy::public())
+        .unwrap();
+    storage_2
+        .create(U64_KEY_2, Value::U64(U64_VALUE_2), &Policy::public())
+        .unwrap();
+
+    // Verify stores/resets do not interfere across namespaces
+    storage_1.reset_and_clear().expect(&format!(
+        "Failed to reset VaultStorage with namespace: {}!",
+        VAULT_NAMESPACE_1
+    ));
+    assert!(storage_1.get(U64_KEY_1).is_err());
+    assert_eq!(
+        storage_2.get(U64_KEY_2).unwrap().value.u64().unwrap(),
+        U64_VALUE_2
+    );
+
+    storage_1
```

Comment:
probably can delete everything after this, so we have a succinct test.

Reply:
For clarity, I've split this test into two tests: 'test_vault_namespace_reset()' and 'test_vault_no_namespace_reset()'. The first checks that a reset inside a namespace only clears local data, and the second checks that a reset outside a namespace clears all data. This should help make things clearer!

Reply:
Can we simplify this a little? no need to be accessing multiple values. I'm a little concerned when I see a huge jump in test code because it might make the tests less maintainable or even thrown away during a refactor.

Reply:
Sure, that makes sense! I've tried to simplify the various tests where I can (without losing test functionality/information). Let me know what you think 😄Thanks!